### DISCUSSION TITLE
save row action saves previous value

### DISCUSSION THREAD

**Discussion by lukuser** [Budibase/budibase#11474] [Help] [answered]
Version 2.9.8 (and 2.7.33)
Self Hosted: Azure ACI
Database: SQL Server

I have a form where I want each field to save to db as soon as it's updated. So in "On change" of each field I add action "Save Row". When I preview and modify value (let's say from "A" to "B") and tab out., I see the confirmation msg "Row saved" but the db doesn't get updated (stays as "A"). At this time in the field visible on the form I still see "B". I try again and change value from "B" to "C" and tab out again. I see the "Row Saved" again, but this time the whole form flashes (reloads) and the value in the box changes back to "B". The database also gets updated to "B". So I get to save only every second change and only to the value of the previous change.
The Save Row data source is the form, not the repeater above the form (if I try to set the ds to repeater, the db doesn't get updated at all - no matter how many times I change the value).

**Comment by avivkri** [comment]
Hello @lukuser,

You might want to use the special binding `{{ Field Value }}` to refer to the current value of the field inside "On change" actions, as explained in the documentation here.
https://docs.budibase.com/docs/text-inputs#on-change

**Reply by lukuser** [reply]
Hello @avivkri, thanks for a response. Sorry. I'm not clear how I can use the {{ Field Value }} on a "On Change.Save Row" action?

**Reply by avivkri** [reply, marked as answer]
No problem. You may use the _Add Column_ option in Save Row action and add both the row id and the column to be updated. The current row id can be retrieved from the Repeater by `{{ Repeater.TableName._id }}`. The current value of the field from the Form can retrieved by `{{ Field Value }}`.

It is like so:
![image](https://github.com/Budibase/budibase/assets/13772138/715e0b4c-d2d5-461d-ae3f-3612265b4064)

**Reply by lukuser** [reply]
Thanks. That worked. For some reason the Add Column button wasn't there on the Add Row. I recreated the component and I got that button back.
